homeassistant/components/sems_portal/sensor.py

```diff
@@ -0,0 +1,278 @@
+"""Sensor for retrieving data for SEMS portal."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CURRENCY_DOLLAR,
+    PERCENTAGE,
+    UnitOfEnergy,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DOMAIN, SemsDataUpdateCoordinator
+
+
+class SemsInformationSensor(SensorEntity):
```

Comment:
Will you add more Platforms later? Then I'd definitely add a common base class, if not I'd still think about it to reduce some of the duplicate code in the inits.

Reply:
Have added a common base class in entity.py (Saw other integrations did this, lmk if this isn't correct practice)